File: go/search/uint32ListCriterion.go
package search

// NewUint32ListCriterion is a convenience function to construct a wrapped new Uint32ListCriterion.
func NewUint32ListCriterion(
	field string,
	list []uint32,
) *Criterion {
	return &Criterion{
		Criterion: &Criterion_Uint32ListCriterion{
			Uint32ListCriterion: &Uint32ListCriterion{
				Field: field,
				List:  list,
			},
		},
	}
}

File: ts/src/search/uint32ListCriterion.ts
import { Criterion } from "./criterion_pb";
import { Uint32ListCriterion } from "./uint32ListCriterion_pb";

/**
 * Convenience function to construct a wrapped new Uint32ListCriterion.
 *
 * @param {string} field - field of list uint32 criterion
 * @param {number[]} list - value of list uint32 criterion
 * @returns {Criterion} Uint32ListCriterion wrapped in Criterion
 *
 * @example
 * // results in the mongo db query {"id": {"$in":[1, 2]}}
 * const uint32ListCriterion = newUint32ListCriterion("id", [1, 2]);
 */
export function newUint32ListCriterion(
  field: string,
  list: number[],
): Criterion {
  return new Criterion().setUint32listcriterion(
    new Uint32ListCriterion().setField(field).setListList(list),
  );
}
